examples/serre_green_naghdi_1d/serre_green_naghdi_soliton_upwind.jl

```diff
@@ -0,0 +1,49 @@
+using OrdinaryDiffEq
+using DispersiveShallowWater
+using SummationByPartsOperators: upwind_operators, periodic_derivative_operator
+using SparseArrays: sparse
+
+###############################################################################
+# Semidiscretization of the Serre-Green-Naghdi equations
+
+equations = SerreGreenNaghdiEquations1D(gravity_constant = 9.81)
+
+# initial_condition_convergence_test needs periodic boundary conditions
+initial_condition = initial_condition_convergence_test
+boundary_conditions = boundary_condition_periodic
+
+# create homogeneous mesh
+coordinates_min = -50.0
+coordinates_max = 50.0
+N = 512
+mesh = Mesh1D(coordinates_min, coordinates_max, N)
+
+# create solver with periodic upwind SBP operators of accuracy order 4
+# (note that only central-type with even order are used)
+accuracy_order = 4
+D1 = upwind_operators(periodic_derivative_operator;
+                      derivative_order = 1,
+                      accuracy_order = accuracy_order - 1,
+                      xmin = xmin(mesh), xmax = xmax(mesh),
+                      N = nnodes(mesh))
+D2 = sparse(D1.plus) * sparse(D1.minus) # FIXME: not used
```

Comment:
Yeah, good question. We could allow passing `nothing` for `D2` and then pass `nothing` here. But I'm not sure if that's ideal.
I also though about that already when thinking about how to implement KdV for example where we would also need a third derivative operator. I'm open for some nice ideas :smile:

Reply:
We decided to allow `nothing` and pass it here, but keep it the fact that there is not default for `D2`. Once we need third derivative order operators we also allow to pass `nothing` for it and `D3` will be `nothing` by default.

Reply:
012303f2917fc0f7ee5aa778976e289924503649